src/commands/refreshSubscriptions.ts

```diff
@@ -0,0 +1,14 @@
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from "vscode-azureextensionui";
+
+export default async function refreshSubscriptions(context: IActionContext, target: any): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    if (cloudExplorer.available) {
+        const commandTarget = cloudExplorer.api.resolveCommandTarget(target);
+
+        if (commandTarget && commandTarget.nodeType === 'resource') {
+            target.provider.treeDataProvider.refresh();
```

Comment:
I'm afraid I'm deeply uncomfortable with the way you get the tree data provider here.  `target` is an opaque value provided by the k8s extension.  The **ONLY** thing we promise you can do with it is call `resolveCommandTarget` on it.  **ANYTHING** else might change at the drop of a hat, in a minor version, without warning. If you want your code to keep working, you can't poke around in the `target` object with the debugger and take a dependency on things that happen to be there on the day you did the poking - you have to stick to the contract.  And the contract (https://github.com/Azure/vscode-kubernetes-tools-api/blob/master/ts/cloudexplorer/v1.ts) says that resolving the target will give you back the resource you originally provided and _nothing else_ (well except the cloud name).

(Really we should treat `target` as type `unknown` rather than type `any` - I wonder if this works with passing it into `resolveCommandTarget`.)

Anyway the upshot is that you need to get the reference the tree provider some other way, either by stashing a copy in every tree node (because the contract promises to hand your tree node back to you in the `cloudResource` member), or by storing it a global, or closing over it, or something.  Sorry.